feat(stdlib): strncpy consistency and add strlcpy

liamHowatt opened the pull request:
### Description of the feature or fix

Following up with the discussion about `strncpy` in #5927

Make `lv_strncpy` consistent with the standard spec.

The standard spec of `strncpy` isn't necessarily better than the one that's implemented in LVGL `lv_string_builtin.c`. The concern is that the behavior of `lv_strncpy` changes depending on the configured `LV_USE_STDLIB_STRING`. We wrap the libc `strncpy` with the modified behavior to match the unique builtin behavior, however we don't wrap the rt-thread implementation and the [rt-thread implementation](https://github.com/RT-Thread/rt-thread/blob/dbf8a26e6c9a8770b2cf708a24e080a81db4f07c/src/klibc/kstring.c#L348) matches the standard spec, i.e. the libc behavior.

I updated the uses of `lv_strncpy` where it's called to ensure a null terminator is added where it's expected.

See the [manpage for strncpy](https://linux.die.net/man/3/strncpy).

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

XuNeo commented:
> I feel like our behavior for `lv_strncpy` is bette

You are looking for `strlcpy`. Let's use `lv_strlcpy` then. The standard is fixed, we cannot do much about it now.

kisvegabor commented:
I didn't know `strlcpy` either, but it's great! I agree to use `lv_strlcpy`. @liamHowatt if you also agree could you update this PR accordingly?

liamHowatt commented:
I agree. I'll add `strlcpy` and use it in most places `strncpy` is.

liamHowatt commented:
Updated.

So it looks like the `strlcpy` is a BSD thing. You need libbsd installed. So I just implemented the clib one in terms of `strlen` and `memcpy` so people using clib string don't need libbsd.

FYI `strlcpy` returns the length of `src`. Something to note.

XuNeo commented on src/others/file_explorer/lv_file_explorer.c:
```diff
@@ -599,7 +599,7 @@ static void show_dir(lv_obj_t * obj, const char * path)
     lv_obj_scroll_to_y(explorer->file_table, 0, LV_ANIM_OFF);
 
     lv_memzero(explorer->current_path, sizeof(explorer->current_path));
```
Remove `lv_memzero(explorer->current_path, sizeof(explorer->current_path));`

XuNeo reviewed (approved):
Thank you!

kisvegabor reviewed (commented):
Thank you! 

I'm not sure we need to keep `lv_strncpy` but let's leave it for backward compatibility.

kdschlosser commented:
@kisvegabor 

I think this is the one that broke the MicroPython CI.

kdschlosser commented:
it looks like this function is wrong. It strips off the trailing `'\0'`.

```c
size_t lv_strlcpy(char * dst, const char * src, size_t dst_size)
{
    size_t i = 0;
    if(dst_size > 0) {
        for(; i < dst_size - 1 && src[i]; i++) {
            dst[i] = src[i];
        }
        dst[i] = '\0';
    }
    while(src[i]) i++;
    return i;
}

```

liamHowatt commented:
That CI was run before this got merged.

kdschlosser commented:
There are only 2 things that got merged in that time frame and this is the only one that would effect the micropython CI

Maybe something got changed over in the micropython binding?

kdschlosser commented:
nothing changed over there either. Gonna have to turn on debugging for the test to see what is happening.
